File: composeApp/src/commonTest/kotlin/com/hereliesaz/hg2gui/commands/main/raw/SwitchOsTest.java
package com.hereliesaz.hg2gui.commands.main.raw;

import org.junit.Test;
import static org.junit.Assert.*;

import com.hereliesaz.hg2gui.commands.ExecutePack;
import com.hereliesaz.hg2gui.managers.SystemContext;

public class SwitchOsTest {

    @Test
    public void testSwitchToMac() throws Exception {
        switchos cmd = new switchos();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"macos"};

        String result = cmd.exec(pack);

        assertNotNull(result);
        assertTrue(result.contains("KERNEL SWITCHED TO: MACOS"));
        assertEquals(SystemContext.OSType.MACOS, SystemContext.getInstance().getOs());
    }

    @Test
    public void testSwitchToWindows() throws Exception {
        switchos cmd = new switchos();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"windows"};

        String result = cmd.exec(pack);

        assertTrue(result.contains("KERNEL SWITCHED TO: WINDOWS"));
        assertEquals(SystemContext.OSType.WINDOWS, SystemContext.getInstance().getOs());
    }

    @Test
    public void testInvalidOs() throws Exception {
        switchos cmd = new switchos();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"templeos"};

        String result = cmd.exec(pack);

        assertTrue(result.contains("NOT FOUND"));
    }
}

File: composeApp/src/commonTest/kotlin/com/hereliesaz/hg2gui/commands/main/raw/SwitchOsTest.kt
package com.hereliesaz.hg2gui.commands.main.raw

import com.hereliesaz.hg2gui.PlatformContext
import com.hereliesaz.hg2gui.commands.ExecutePack
import com.hereliesaz.hg2gui.managers.SystemContext
import org.junit.Assert.*
import org.junit.Test

class SwitchOsTest {

    private val mockPack = object : ExecutePack() {
        override val context: PlatformContext = object : PlatformContext {
            override fun getString(resId: Int): String = ""
            override fun getString(resId: Int, vararg args: Any): String = ""
        }
        override fun getPrefsSave(): Any? = null
        override fun getLaunchInfo(): Any? = null
    }

    @Test
    fun testSwitchToMac() {
        val cmd = switchos()
        mockPack.args = arrayOf("macos")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)

        assertNotNull(result)
        assertTrue(result.contains("KERNEL SWITCHED TO: MACOS"))
        assertEquals(SystemContext.OSType.MACOS, SystemContext.getInstance().os)
    }

    @Test
    fun testSwitchToWindows() {
        val cmd = switchos()
        mockPack.args = arrayOf("windows")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)

        assertTrue(result.contains("KERNEL SWITCHED TO: WINDOWS"))
        assertEquals(SystemContext.OSType.WINDOWS, SystemContext.getInstance().os)
    }

    @Test
    fun testInvalidOs() {
        val cmd = switchos()
        mockPack.args = arrayOf("templeos")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)

        assertTrue(result.contains("NOT FOUND"))
    }
}
